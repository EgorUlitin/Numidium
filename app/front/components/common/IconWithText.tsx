import * as React from 'react'

import Icon from './Icon'

import IconEnum from 'util/IconEnum'


interface Props {
    icon?: IconEnum
    text: string
}

export default (props: Props) =>
    <React.Fragment>
        {props.icon && <Icon type={props.icon} />}
        <span>{props.text}</span>
    </React.Fragment>
