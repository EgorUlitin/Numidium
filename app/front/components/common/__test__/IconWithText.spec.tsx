import * as React from 'react'
import * as renderer from 'react-test-renderer'

import IconWithText, { IconType } from '../IconWithText'

test('`Icon` renders correctly', () => {
    const tree = renderer
        .create(<IconWithText icon={IconType.ANDROID} text={'My favorite text'} />)
        .toJSON()

    expect(tree).toMatchSnapshot()
})
