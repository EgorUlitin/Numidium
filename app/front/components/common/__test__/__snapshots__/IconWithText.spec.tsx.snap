// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Icon\` renders correctly 1`] = `
Array [
  <i
    className="anticon anticon-android"
    onClick={undefined}
    style={undefined}
  />,
  <span>
    My favorite text
  </span>,
]
`;
